warehouse/db.go

```diff
@@ -0,0 +1,50 @@
+package warehouse
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+
+	warehouseutils "github.com/rudderlabs/rudder-server/warehouse/utils"
+)
+
+// DB encapsulate interactions of warehouse operations
+// with the database.
+type DB struct {
```

Comment:
Can we expose an interface here instead of calling it from a struct? wdyt
Later we can easily mock the interface.